test/test.js

```diff
@@ -4,20 +4,38 @@ import { jams, read } from '../jams.js'
 
 import { readFileSync, readdirSync } from 'fs'
 
-test('passing files and their JSON equivalents', t=> {
-    readdirSync('./test/pass').forEach(filename => {
-        const extension = filename.split('.').slice(-1)
-        if (extension == "jams"){ // only read JAMS, look for json if jams exists
-            const jams_o = jams(readFileSync(`./test/pass/${filename}`))
-            const jsonfile = readFileSync(`./test/pass/${filename}`.replace('jams', 'json'))
-            if (jsonfile) {
-                const json_o = JSON.parse(jsonfile)
-                for (const key in json_o) {
-                    t.ok(jams_o[key])
-                    t.deepEqual(json_o[key], jams_o[key])
-                }
-            }
+test('passing files and their JSON equivalents', _=> {
+    const subpath = './test/pass'
+    readdirSync(subpath).forEach(filename => {
+        const extension = filename.split('.').slice(-1)[0]
+	// Only read JAMS files, and we require a corresponding JSON verison.
+        if (extension === "jams") {
+	    test(filename, t=> {
+		const jamspath = `${subpath}/${filename}`
+	        const jams_o = jams(readFileSync(jamspath))
+	        const jsonfile = readFileSync(jamspath.replace('jams', 'json'))
+	        const json_o = JSON.parse(jsonfile)
+	        t.deepEqual(Object.keys(jams_o), Object.keys(json_o))
+	        for (const key in json_o) {
```

Comment:
It's a bug when the JSON doesn't have keys and JAMS does - actually hit it when I was playing around.